starlark/int_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package starlark
+
+import (
+	"math"
+	"testing"
+)
+
+func TestIntOpts(t *testing.T) {
+	f := maybeBig
+	for i, test := range []Int{
+		// Add
+		f(math.MaxInt32).Add(f(1)),
+		f(math.MinInt32).Add(f(-1)),
+		// Mul
+		f(math.MaxInt32).Mul(f(math.MaxInt32)),
+		f(math.MinInt32).Mul(f(math.MinInt32)),
+		f(math.MinInt32).Mul(f(-1)),
+		// Div
+		f(math.MinInt32).Div(f(-1)),
+		f(1 << 31).Div(f(2)),
+		// And
+		f(math.MaxInt32).And(f(math.MaxInt32)),
+		f(math.MinInt32).And(f(math.MinInt32)),
+		f(1 << 33).And(f(1 << 32)),
+		// Mod
+		f(1 << 32).Mod(f(2)),
+		// Or
+		f(1 << 32).Or(f(0)),
+		f(math.MaxInt32).Or(f(0)),
+		f(math.MinInt32).Or(f(math.MinInt32)),
+		// Xor
+		f(math.MinInt32).Xor(f(-1)),
+		// Not
+		f(math.MinInt32).Not(),
```

Comment:
I'm a bit surprised about this one. ~ -0x80000000 = 0x7fffffff. Both are int32 bits.

Perhaps this is a consequence of using < not <= (see above).

Reply:
I exposed big.Ints Format on Int and added %x printing, is there a better way to do that?